app/connectors/PSRConnector.scala

```diff
@@ -44,28 +44,35 @@ class PSRConnector @Inject()(appConfig: FrontendAppConfig, http: HttpClient) {
     s"$baseUrl/pension-scheme-return/psr/versions/$pstr?startDate=$startDate"
 
   def submitPsrDetails(
-    psrSubmission: PsrSubmission
-  )(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Either[String, Unit]] =
+    psrSubmission: PsrSubmission,
+    userName: String,
+    schemeName: String
+  )(implicit headerCarrier: HeaderCarrier, ec: ExecutionContext): Future[Either[String, Unit]] = {
+    val hc: HeaderCarrier = buildHeaders(userName, schemeName, headerCarrier)
```

Comment:
UserName is PII and there is a chance it will end up in some access logs as header value. Can it be encrypted? We already use encryption for the email callback url parameters in EmailResponseController.